src/main.rs

```diff
@@ -403,6 +407,54 @@ fn create_app<'a>(
 }
 
 fn start_vmm(cmd_arguments: ArgMatches) -> Result<Option<String>, Error> {
+    // Can't test for "vm-config" group as some have default values. The kernel (or tdx if enabled)
+    // is the only required option for booting the VM.
+    #[cfg(feature = "tdx")]
+    let tdx_or_kernel_present =
+        cmd_arguments.is_present("kernel") || cmd_arguments.is_present("tdx");
+    #[cfg(not(feature = "tdx"))]
+    let tdx_or_kernel_present = cmd_arguments.is_present("kernel");
+    let vm_config = if tdx_or_kernel_present {
+        let vm_params = config::VmParams::from_arg_matches(&cmd_arguments);
+        Some(config::VmConfig::parse(vm_params).map_err(Error::ParsingConfig)?)
+    } else {
+        None
+    };
+    #[cfg(feature = "amx")]
+    if let Some(vm_config) = vm_config {
```

Comment:
I agree both the parsing and the validation should happen in the `CpuManager`'s code since this is a `--cpus` parameter.

Reply:
@sboeuf I think I'm a little confused, doesn't the parsing for the --cpus option need to happen in `vmm::config`?

Reply:
@rbradford Okay, I had put it here largely because I figured this would be good to immediately fail on rather than spin up the API when the vmm wasn't successfully going to start anyway. Getting the migration bits though is certainly more appealing.

Reply:
> @sboeuf I think I'm a little confused, doesn't the parsing for the --cpus option need to happen in `vmm::config`?

I don't think he meant "parsing" per se but more "interpretation". The parsing should happen in `vmm::config`

Reply:
Yep sorry for the wording, I meant the interpretation :)